src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -219,31 +219,61 @@ private static bool IsRunProjectCommand(ILaunchProfile profile)
             else
             {
                 // If the working directory is not rooted we assume it is relative to the project directory
-                if (Path.IsPathRooted(resolvedProfile.WorkingDirectory))
+                workingDir = resolvedProfile.WorkingDirectory.Replace("/", "\\");
+                if (Path.GetPathRoot(workingDir) == "\\")
                 {
-                    workingDir = resolvedProfile.WorkingDirectory;
+                    // Root of current drive, no drive specified
+                    workingDir = TheFileSystem.GetFullPath(workingDir);
                 }
-                else
+                else if (!Path.IsPathRooted(workingDir))
                 {
-                    workingDir = Path.GetFullPath(Path.Combine(projectFolder, resolvedProfile.WorkingDirectory));
+                    workingDir = TheFileSystem.GetFullPath(Path.Combine(projectFolder, workingDir));
                 }
```

Comment:
It seems like this entire block and below could be much simpler, like so:

``` C#
workingDir = TheFileSystem.GetFullPath(Path.Combine(projectFolder, resolvedProfile.WorkingDirectory));
```

This covers the following cases:
```
Project     Working                     Result
---------------------------------------------------------
(C:\Foo     Bar       ) ==              C:\Foo\Bar
(C:\Foo     Bar\Baz   ) ==              C:\Foo\Bar\Baz
(C:\Foo     \Bar      ) ==              c:\Bar
(C:\Foo     \Bar\Baz  ) ==              c:\Bar\Baz
(C:\Foo     ..\Bar    ) ==              C:\Bar
(C:\Foo     ..\Bar\Baz) ==              C:\Bar\Baz
(C:\Foo     E:\Bar    ) ==              E:\Bar
(C:\Foo     E:\Bar\Baz) ==              E:\Bar\Baz
```

Reply:
If the value specified is just the filename then it needs to be relative to the current working directory, not the project. But otherwise, yes that does simplify things quite a bit since the combination of PathCombine & GetFullPath does the right thing.

Reply:
Yes, that is much simpler. Thanks.